sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -379,6 +381,33 @@ class Analyzer(catalog: Catalog,
     }
   }
 
+  object ResolveSortAlias extends Rule[LogicalPlan] {
+    def apply(plan: LogicalPlan): LogicalPlan = plan transform {
+      case s @ Sort(ordering, global, a @ Aggregate(grouping, aggs, child))
+          if !s.resolved && a.resolved =>
+        val missingInAggs = new ArrayBuffer[NamedExpression]()
+        val aliasedOrdering = ordering.zipWithIndex.map{
+          case (o, i) => {
+            o.child match {
+              case aggOrSub @ (_: AggregateExpression | _: Substring) => {
```

Comment:
Okay, but that sounds like there if a flaw in the logic here and this is just a hack to workaround a single case of this flaw.